blocksync/blocksync.go

```diff
@@ -195,3 +211,19 @@ func (bs *blockSyncer) ProcessSyncRequest(ctx context.Context, peer peerstore.Pe
 	}
 	return nil
 }
+
+func (bs *blockSyncer) syncStageChecker() {
+	syncStageHeight := atomic.LoadUint64(&bs.syncStageHeight)
+	if syncStageHeight > 0 {
+		atomic.StoreUint64(&bs.syncSpeed, (bs.bc.TipHeight()-syncStageHeight)/5)
+
+	}
+	atomic.StoreUint64(&bs.syncStageHeight, bs.bc.TipHeight())
+}
+
+// SyncStatus report block sync status
+func (bs *blockSyncer) SyncStatus() string {
+	targetHeight := bs.TargetHeight()
+	syncSpeed := atomic.LoadUint64(&bs.syncSpeed)
+	return fmt.Sprintf("syncSpeed: %d/s, targetHeight: %d", syncSpeed, targetHeight)
```

Comment:
here check bs.syncBlockIncrease
if == 1
  return "synced to blockchain tip, height = 8123456"
else
  // calculate the speed as float64
  return "sync in progress, speed: 53 blocks/sec, height = 7891234"

Reply:
also no need to give targetHeight, give current height